src/Remote/AMQP/AMQPConsumer.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Onliner\CommandBus\Remote\AMQP;
+
+use Generator;
+use Onliner\CommandBus\Dispatcher;
+use Onliner\CommandBus\Remote\Consumer;
+use Onliner\CommandBus\Remote\Envelope;
+use PhpAmqpLib\Channel\AMQPChannel;
+use PhpAmqpLib\Exception\AMQPConnectionClosedException;
+use PhpAmqpLib\Message\AMQPMessage;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Throwable;
+
+final class AMQPConsumer implements Consumer
+{
+    /**
+     * @var Connector
+     */
+    private $connector;
+
+    /**
+     * @var ExchangeOptions
+     */
+    private $options;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var array<string>
+     */
+    private $listen = [];
+
+    /**
+     * @var bool
+     */
+    private $running = false;
+
+    /**
+     * @param Connector            $connector
+     * @param ExchangeOptions|null $options
+     * @param LoggerInterface|null $logger
+     */
+    public function __construct(Connector $connector, ExchangeOptions $options = null, LoggerInterface $logger = null)
+    {
+        $this->connector = $connector;
+        $this->options   = $options ?? ExchangeOptions::default();
+        $this->logger    = $logger ?? new NullLogger();
+    }
+
+    /**
+     * @param string $queue
+     *
+     * @return void
+     */
+    public function listen(string $queue): void
+    {
+        $this->listen[] = $queue;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function run(Dispatcher $dispatcher): void
+    {
+        $channel = $this->connector->channel();
+        $handler = function (AMQPMessage $message) use ($dispatcher) {
+            try {
+                $this->handle($message, $dispatcher);
+            } catch (Throwable $error) {
+                $this->logger->error($error->getMessage());
```

Comment:
```suggestion
                $this->logger->error((string) $error);
```

очень часто у экзепшена нет сообщения, ну и не понятно как он появился
предлагаю выводить с трейсом ошибку